server/controllers/calls.js

```diff
@@ -107,6 +108,33 @@ function afterPutCallAttempt(res, outcome, contact_id, attempt_num, campaign_id)
   }
   return res.status(200).json({ message: 'call log successfully updated' });
 }
+
+function handleHangUpFlow(res, user_id, call_id, newStatus) {
+  return usersService.getUserById({ id: user_id })
+  .then((userObj) => {
+    const { call_sid: userCallSid } = userObj.attributes;
+    console.log(userObj.attributes);
+    console.log(userCallSid, 'inside hang up handler');
+    hangUpContactCall(userCallSid, user_id)
+    .then(() => {
+      callsService.updateCallStatus({ id: call_id, status: newStatus })
+      .then((updatedCall) => {
+        const updateCallSuccess = `call status updated to ${newStatus}`;
+        return res.status(200)
+          .json({ message: updateCallSuccess, call: updatedCall });
+      })
+      .catch((err) => {
+        const updateCallError = `Could not update call status to ${newStatus}: ${err}`;
+        return res.status(500).json({ message: updateCallError });
+      });
+    })
+    .catch((err) => {
+      const hangUpCallError = `Could not hang up call from twilio client to contact: ${err}`;
+      return res.status(500).json({ message: hangUpCallError });
+    });
+  })
+  .catch(err => err);
```

Comment:
Perhaps return a 404 since we couldn't find the user?

Reply:
Good suggestion, done/